python/dgl/graphbolt/feature_store.py

```diff
@@ -134,3 +140,73 @@ def update(self, value: torch.Tensor, ids: torch.Tensor = None):
                 f"but got {ids.shape[0]} and {value.shape[0]}."
             )
             self._tensor[ids] = value
+
+
+class OnDiskFeatureData(pydantic.BaseModel):
+    r"""The description of an on-disk feature."""
+    name: str
+    format: str
```

Comment:
try with enum like this https://github.com/dmlc/dgl/blob/8adb53bb1c7cef11efb3d769455521757e583f97/python/dgl/graphbolt/dataset.py#L69 /